xblock/test/test_core.py

```diff
@@ -1039,6 +1039,13 @@ class TestBlock(XBlock):
         """An empty XBlock for testing"""
         pass
 
+    # FIXME: This test is fragile - fails in python27 if a warning it is expecting is emitted before it is executed
```

Comment:
@robrap FYI.

Reply:
The warnings module has different default filter behavior for different warning categories across different versions of python. Most likely DeprecationWarning is set to ["once"](https://docs.python.org/2/library/warnings.html#the-warnings-filter) in this case.

I think this can be worked around by calling `warnings.resetwarnings()` or `warnings.simplefilter("always")` in the test setup.

Reply:
TIL.  Thanks @bdero

Reply:
@jcdyer @bdero No guys, it's [not that simple](https://github.com/edx/XBlock/blob/master/xblock/test/tools.py#L118).

Reply:
@e-kolpakov Have you tried sliding a `warnings.resetwarnings()` in there before the catch_warnings line? It's totally possible there's a subtle behavior difference between different versions of python (or a bug in the specific release being used), so hopefully playing around with the warning options will yield some good information on the problem.

Reply:
@bdero Yes, I've tried (and double checked just now) spamming with `resetwarnings` and `filter("always")` with different combinations and at different locations - no help.

Unfortunately, I don't have capacity right now to further investigate this issue, so I'd suggest we just wrap it up as is.

@robrap It's not a flaky test - it is a stable test that's easy to break as it depends on the order of tests executed. In short, it only passes now because `test_core` is run before any other test that could trigger the deprecation warning. If I understand right, nose runs tests alphabetically, so test_core is run very early - only asides and now completion is run earlier. Asides code does not have anything to do with creating XBlocks (where the deprecation warning is set), and completion also doesn't use the deprecated feature - but only after I figured out what's wrong with it - most of the other tests are still using it.

Reply:
@e-kolpakov @robrap I suspect that sometime reasonably soon we'll want to switch this repository to use pytest instead of nose, and make a concerted effort to fix any dependencies between the tests so they can be run in random order.  If the tests are stable until the order changes, I think it's probably ok to leave it as is until we tackle that.  pytest has some dedicated functionality for testing warnings which may prove useful when we attempt that.

Reply:
Thanks @e-kolpakov.  You may want to add an additional note regarding @jmbowman's response in the comment if you think it would be helpful to someone coming across this.  Or, you could leave a link to this discussion regarding how and when it might get fixed.